src/test/java/com/artipie/rpm/pkg/PrimaryOutputTest.java

```diff
@@ -72,4 +72,58 @@ void createsPrimaryForAbc(@TempDir final Path temp) throws IOException {
             )
         );
     }
+
+    @Test
+    void checkFile(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("primary.xml");
+        try (PackageOutput.FileOutput primary = new PrimaryOutput(res)) {
+            primary.start();
+            MatcherAssert.assertThat(
+                primary.file(),
+                new IsEqual<>(res)
+            );
+        }
+    }
+
+    @Test
+    void checkTag(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("primary.xml");
+        try (PackageOutput.FileOutput primary = new PrimaryOutput(res)) {
+            primary.start();
+            MatcherAssert.assertThat(
+                primary.tag(),
+                new IsEqual<>("metadata")
+            );
+        }
+    }
+
+    @Test
+    void createsPrimaryForLibdeflt(@TempDir final Path temp) throws IOException {
+        final Path res = temp.resolve("primary.xml");
+        try (PackageOutput.FileOutput primary = new PrimaryOutput(res)) {
+            primary.start();
+            final Path rpm =
+                Paths.get("src/test/resources-binary/libdeflt1_0-2020.03.27-25.1.armv7hl.rpm");
+            primary.accept(
+                new FilePackage.Headers(new FilePackageHeader(rpm).header(), rpm, Digest.SHA256)
+            );
+        }
+        MatcherAssert.assertThat(
+            Files.readAllBytes(res),
+            CompareMatcher.isIdenticalTo(
+                Files.readAllBytes(
+                    Paths.get("src/test/resources-binary/repodata/libdeflt-primary.xml")
+                )
+            ).ignoreWhitespace()
+            .ignoreElementContentWhitespace()
+            .normalizeWhitespace()
+            .withNodeFilter(
+                node -> !"file".equals(node.getLocalName())
```

Comment:
@paulodamaso are you sure we need to filter `file` nodes? As I can see, example xml does not have `file` list, xml we generate has?